tests/v1/shutdown/test_startup_error.py

```diff
@@ -0,0 +1,88 @@
+# SPDX-License-Identifier: Apache-2.0
+"""Test that we handle a startup Error and shutdown."""
+
+import pytest
+
+from tests.utils import wait_for_gpu_memory_to_clear
+from vllm import LLM
+from vllm.distributed import get_tensor_model_parallel_rank
+from vllm.engine.arg_utils import AsyncEngineArgs
+from vllm.model_executor.models.llama import LlamaForCausalLM
+from vllm.utils import cuda_device_count_stateless
+from vllm.v1.engine.async_llm import AsyncLLM
+
+
+def evil_forward(self, *args, **kwargs):
+    """Evil forward method that raise an exception."""
+
+    if get_tensor_model_parallel_rank() == 0:
+        raise Exception("Simulated Error in startup!")
+
+    return self.model(*args, **kwargs, intermediate_tensors=None)
+
+
+MODELS = [
+    "meta-llama/Llama-3.2-1B",  # Raises on first fwd pass.
+    "mistralai/Mixtral-8x22B-Instruct-v0.1"  # Causes OOM.
```

Comment:
Is this going to download the model and get a real OOM? From a quick look, it doesn't look like this is used elsewhere, so that'd be a net-new model to download during tests? If so, that doesn't seem worth the cost, especially given how unreliable HF has been in CI lately. Maybe I'm misunderstanding, though!

Reply:
Good idea.

I do think that it is important to flex both the cases here since there is a subtle difference:
- "meta-llama/Llama-3.2-1B",  # Raises on first fwd pass.  happens during the profiling (after IPC mechanisms are created)
- "mistralai/Mixtral-8x22B-Instruct-v0.1"  # Causes OOM. happens during the weight loading (before IPC mechanisms are setup)

I will instead do a monkeypatch to raise an error on load_weights for case

Reply:
monkeypatch sounds good if the error encountered is clear enough!